### PR TITLE
NGSTACK-325 Search spellcheck suggestion

### PR DESCRIPTION
**Pull request by RandyCupic**
This PR enables spellcheck for search and provides the suggested term in SearchController.

### CODE REVIEW

File: bundle/API/Search/Criterion/FullText.php [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+use eZ\Publish\API\Repository\Values\Content\Query\Criterion\FullText as BaseFullText;
+use \Netgen\EzPlatformSearchExtra\API\Values\Content\Query\Criterion\FulltextSpellcheck;
+use Netgen\EzPlatformSearchExtra\API\Values\Content\SpellcheckQuery;
+
+class FullText extends BaseFullText implements FulltextSpellcheck
+{
+    /**
+     * Gets query to be used for spell check.
+     *
+     * @return \Netgen\EzPlatformSearchExtra\API\Values\Content\SpellcheckQuery
```
Comment (emodric) [inline bundle/API/Search/Criterion/FullText.php:16]: No need for `@return` since you have a return type hint anyway.

File: bundle/API/Search/Criterion/FullText.php [hunk 2]
```diff
@@ -0,0 +1,26 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Netgen\Bundle\SiteBundle\API\Search\Criterion;
+
+use eZ\Publish\API\Repository\Values\Content\Query\Criterion\FullText as BaseFullText;
+use \Netgen\EzPlatformSearchExtra\API\Values\Content\Query\Criterion\FulltextSpellcheck;
```
Comment (emodric) [inline bundle/API/Search/Criterion/FullText.php:8]: Extra `\` at the start.

File: bundle/Controller/SearchController.php [hunk 3]
```diff
@@ -4,14 +4,25 @@
 
 namespace Netgen\Bundle\SiteBundle\Controller;
 
+use eZ\Publish\API\Repository\Exceptions\NotFoundException;
+use Netgen\Bundle\SiteBundle\Core\Search\SuggestionResolver;
 use Netgen\EzPlatformSiteApi\Core\Site\Pagination\Pagerfanta\FindAdapter;
 use Pagerfanta\Pagerfanta;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
-use function trim;
```
Comment (emodric) [inline bundle/Controller/SearchController.php:11]: Why this change? You probably have an older version of PHP CS Fixer that does not understand the rule for this.

File: bundle/Core/Search/SuggestionResolver.php [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+use eZ\Publish\API\Repository\Values\Content\Query;
+use eZ\Publish\Core\Base\Exceptions\NotFoundException;
+use Netgen\EzPlatformSearchExtra\API\Values\Content\Search\Suggestion;
+
+class SuggestionResolver
+{
+    /**
+     * Get suggested search term based on returned spell check suggestions.
+     *
+     * @param \eZ\Publish\API\Repository\Values\Content\Query $query
```
Comment (emodric) [inline bundle/Core/Search/SuggestionResolver.php:16]: No need for `@param` and `@return`.

File: bundle/Core/Search/SuggestionResolver.php [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+        $replace = [];
+
+        foreach ($suggestion->getOriginalWords() as $originalWord) {
+            $replace[] = $suggestion->getSuggestionsByOriginalWord($originalWord)[0]->suggestedWord;
+        }
+
+        $suggestedTerm = str_replace($search, $replace, $searchTerm);
+
+        if ($searchTerm === $suggestedTerm) {
+            throw new NotFoundException('suggestion', 'for search term {'.$searchTerm.'}');
```
Comment (emodric) [inline bundle/Core/Search/SuggestionResolver.php:36]: Why `{...}`? Just use double quotes, e.g.: `sprintf('for search term "%s"', $searchTerm)`

**Comment by emodric** [comment]
Besides the nitpicks, looks good :+1:

File: bundle/API/Search/Criterion/FullText.php (hunk 1 above)
Comment (RandyCupic) [reply bundle/API/Search/Criterion/FullText.php]: @emodric Fixed in https://github.com/netgen/site-bundle/pull/14/commits/d29931259d3c7e08abef9ea768050720e9e2182b.

File: bundle/API/Search/Criterion/FullText.php (hunk 2 above)
Comment (RandyCupic) [reply bundle/API/Search/Criterion/FullText.php]: @emodric Fixed in https://github.com/netgen/site-bundle/pull/14/commits/f07177aaf93e125bae0700733c66fc296c8ec9c7.

File: bundle/Controller/SearchController.php (hunk 3 above)
Comment (RandyCupic) [reply bundle/Controller/SearchController.php]: @emodric I haven't used the CS fixer,  I've commited that accidentally :) Fixed in https://github.com/netgen/site-bundle/pull/14/commits/e903d27b8ba5c72ca09b3e350b37157ed34c5070.

File: bundle/Core/Search/SuggestionResolver.php (hunk 5 above)
Comment (RandyCupic) [reply bundle/Core/Search/SuggestionResolver.php]: @emodric C/P from another implementation from project :) Fixed in https://github.com/netgen/site-bundle/pull/14/commits/3e8c09c8641d8f9c868449a70254273b20d7ad9e

File: bundle/Core/Search/SuggestionResolver.php (hunk 4 above)
Comment (RandyCupic) [reply bundle/Core/Search/SuggestionResolver.php]: @emodric Fixed in https://github.com/netgen/site-bundle/pull/14/commits/3e8c09c8641d8f9c868449a70254273b20d7ad9e.

**Review by emodric — APPROVED** [review]

**Comment by emodric** [comment]
Thanks @RandyCupic !
